android/java/org/chromium/chrome/browser/app/BraveActivity.java

```diff
@@ -500,18 +500,19 @@ public WalletModel getWalletModel() {
 
     private void maybeHasPendingUnlockRequest() {
         assert mKeyringService != null;
-        mKeyringService.hasPendingUnlockRequest(pending -> {
-            if (pending) {
-                BraveToolbarLayoutImpl layout = getBraveToolbarLayout();
-                if (layout != null) {
-                    layout.showWalletPanel();
-                }
+        mKeyringService.hasPendingUnlockRequest(
+                pending -> {
+                    if (pending) {
+                        BraveToolbarLayoutImpl layout = getBraveToolbarLayout();
+                        if (layout != null) {
+                            layout.showWalletPanel();
+                        }
 
-                return;
-            }
-            maybeShowPendingTransactions();
-            maybeShowSignTxRequestLayout();
```

Comment:
Maybe I'm not grasping the whole picture here, but what happened to the ETH pending transactions? The method `maybeShowSignTxRequestLayout()` was in charge of processing both (SOL + ETH pending transaction requests).

Reply:
`maybeShowSignTxRequestLayout` -> `SIGN_TRANSACTION`
`maybeShowSignAllTxRequestLayout` -> `SIGN_ALL_TRANSACTIONS`
were both in charge of SOL transactions only. 

Now both cases are covered by a single type
`maybeShowSignSolTransactionsRequestLayout` -> `SIGN_SOL_TRANSACTIONS`

Reply:
Okay, then there's something sketchy going on: the method `maybeShowSignSolTransactionsRequestLayout` is called for all transactions (ETH too). ETH pending transactions need it otherwise the panel won't show up.

Here is a quick recording. Precondition: 1 ETH pending transaction. Observe how the breakpoint for the method `maybeShowSignSolTransactionsRequestLayout` is hit and processed anyway.

[Screencast from 13-09-2024 13:53:02.webm](https://github.com/user-attachments/assets/f8cab8a1-064c-445a-95a0-eac7f1fd2d40)

Here is another recording, same ETH pending transaction, but the method `maybeShowSignSolTransactionsRequestLayout` was commented. Result: the wallet panel is never displayed.

[Screencast from 13-09-2024 13:59:18.webm](https://github.com/user-attachments/assets/236036b6-e513-43ae-b4ac-fd2106df4065)

Let's continue on DMs.